File: ls_treso/ls_treso/report/mouvement_par_nature_des_op/mouvement_par_nature_des_op.js
// Copyright (c) 2023, Kossivi Amouzou and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Mouvement par Nature des OP"] = {
	"filters": [
		{
			fieldname:"date_debut",
			label: __("Date Début"),
			fieldtype: "Date",
			default: frappe.datetime.month_start(),
			reqd: 1
		},
		{
			fieldname:"date_fin",
			label: __("Date Fin"),
			fieldtype: "Date",
			default: frappe.datetime.month_end(),
			reqd: 1
		},
		{
			fieldname: "nature",
			label: __("Nature"),
			fieldtype: "Link",
			options: "Nature Operations",
		},
		/*{
			fieldname: "account_type",
			label: __("Account Type"),
			fieldtype: "Select",
			options: [
				{ "value": "Accumulated Depreciation", "label": __("Accumulated Depreciation") },
				{ "value": "Asset Received But Not Billed", "label": __("Asset Received But Not Billed") },
				{ "value": "Bank", "label": __("Bank") },
				{ "value": "Cash", "label": __("Cash") },
				{ "value": "Chargeble", "label": __("Chargeble") },
				{ "value": "Capital Work in Progress", "label": __("Capital Work in Progress") },
				{ "value": "Cost of Goods Sold", "label": __("Cost of Goods Sold") },
				{ "value": "Depreciation", "label": __("Depreciation") },
				{ "value": "Equity", "label": __("Equity") },
				{ "value": "Expense Account", "label": __("Expense Account") },
				{ "value": "Expenses Included In Asset Valuation", "label": __("Expenses Included In Asset Valuation") },
				{ "value": "Expenses Included In Valuation", "label": __("Expenses Included In Valuation") },
				{ "value": "Fixed Asset", "label": __("Fixed Asset") },
				{ "value": "Income Account", "label": __("Income Account") },
				{ "value": "Payable", "label": __("Payable") },
				{ "value": "Receivable", "label": __("Receivable") },
				{ "value": "Round Off", "label": __("Round Off") },
				{ "value": "Stock", "label": __("Stock") },
				{ "value": "Stock Adjustment", "label": __("Stock Adjustment") },
				{ "value": "Stock Received But Not Billed", "label": __("Stock Received But Not Billed") },
				{ "value": "Tax", "label": __("Tax") },
				{ "value": "Temporary", "label": __("Temporary") },
			],
		},*/

	]
};

File: ls_treso/ls_treso/report/mouvement_par_nature_des_op/mouvement_par_nature_des_op.py
# Copyright (c) 2023, Kossivi Amouzou and contributors
# For license information, please see license.txt

import frappe
from frappe import _

def execute(filters=None):
	filters = frappe._dict(filters or {})
	columns = get_columns(filters)
	data = get_data(filters)
	return columns, data


def get_columns(filters):
	columns = [
		{ "label": _("N° Pièce"), "fieldtype": "Data",	"fieldname": "name", "width": 100, },
		{ "label": _("Caisse"), "fieldtype": "Link", "fieldname": "caisse", "options": "Caisse", "width": 100, },
		{ "label": _("Bénéficiaire"), "fieldtype": "Data", "fieldname": "remettant", "width": 100, },
		{ "label": _("Désignation"), "fieldtype": "Data", "fieldname": "designation", "width": 100, },
		{ "label": _("Montant"), "fieldtype": "Currency", "fieldname": "montant", "options": "devise", "width": 100, },
	]
	return columns


def get_data(filters):

	#frappe.msgprint(str(filters))

	data = frappe.db.sql(
        """
        SELECT o.date, 
		o.name, 
		o.remettant,
		o.designation,
		o.montant,
		o.devise, o.caisse
		FROM (
			SELECT *
			FROM tabEncaissement
			UNION
			SELECT *
			FROM tabDecaissement
			) o 
		INNER JOIN `tabDetails Operation de Caisse` d on o.name = d.parent
		WHERE o.date >= %(date_debut)s AND o.date <= %(date_fin)s AND d.nature_operations LIKE %(nature)s
        """,{"date_debut": filters.date_debut, "date_fin": filters.date_fin, "nature": filters.nature if filters.nature !=  None else "%"}, as_dict = 1
    )

	return data

